ethpm/package.py

```diff
@@ -80,6 +74,40 @@ def get_contract_factory(self, name: ContractName, w3: Web3 = None) -> Contract:
         contract_factory = current_w3.eth.contract(**contract_kwargs)
         return contract_factory
 
+    def get_contract_instance(
+        self, name: ContractName, address: Address, w3: Web3 = None
```

Comment:
I'm forgetting, why do we accept `w3` as an argument to all of these methods?  Why don't we just always use `self.w3`?

Reply:
@davesque Well, we don't require that a `web3` when instantiating a `Package`. We provide a `Package.set_default_w3` to set the `web3` after creating a `Package`. @pipermerriam correct me if i'm wrong, but I think we want to leave the ability to pass in a `web3` to `Package.get_contract_factory` and `Package.get_contract_instance` - as it provides flexibility for the user to switch between chains - rather than strictly using the default `w3` set on `Package` for all web3 tasks.

Reply:
> I think we want to leave the ability to pass in a w3 to `Package.get_contract_factory` and `Package.get_contract_instance` - as it provides flexibility for the user to switch between chains

You are correct that when we originally talked about this that was the plan, however, I don't think it was the right choice.  It's probably easier for us to write the code and easier for consumers of the API to just require a web3 instance up front and then to not have to worry about it from there on.